packages/firebase_auth/firebase_auth_platform_interface/test/mock.dart

```diff
@@ -70,22 +72,35 @@ Future<void> simulateEvent(String name, Map<String, dynamic> user) async {
   );
 }
 
-Future<void> testExceptionHandling(String type, Function testMethod) async {
-  try {
-    await testMethod();
-  } on FirebaseAuthException catch (_) {
-    if (type == 'PLATFORM' || type == 'EXCEPTION') {
-      return;
-    }
-    fail(
-        'testExceptionHandling: ${testMethod} threw unexpected FirebaseAuthException');
-  } catch (e) {
-    fail('testExceptionHandling: ${testMethod} threw invalid exception ${e}');
-  }
+Future<void> testExceptionHandling(
+  String type,
+  void Function() testMethod,
+) async {
+  await expectLater(
```

Comment:
Same behaviour, but much better error message

There are many tests that should be changed to follow this pattern